NitroxPatcher/Patches/Dynamic/Constructable_Construct_Patch.cs

```diff
@@ -1,62 +1,185 @@
-                BaseGhost baseGhost = constructableBase.gameObject.GetComponentInChildren<BaseGhost>();
-                if (baseGhost != null)
-                {
-                    lastTargetBase = baseGhost.TargetBase;
-                    lastTargetBaseOffset = baseGhost.TargetOffset;
-                }
-
-                lastFace = baseGhost switch
-                {
-                    BaseAddFaceGhost { anchoredFace: { } } baseAddFaceGhost => baseAddFaceGhost.anchoredFace.Value,
-                    BaseAddModuleGhost { anchoredFace: { } } baseAddModuleGhost => baseAddModuleGhost.anchoredFace.Value,
-                    _ => lastFace
-                };
+                CoroutineHost.StartCoroutine(BroadcastObjectBuilt(constructableBase, entityId));
+                return;
             }
-            else
+            IEnumerator postSpawner = BuildingPostSpawner.ApplyPostSpawner(constructable.gameObject, entityId);
+            // Can be null if no post spawner is set for the constructable's techtype
+            if (postSpawner != null)
             {
-                lastTargetBase = null;
-                lastTargetBaseOffset = default(Int3);
+                CoroutineHost.StartCoroutine(postSpawner);
             }
+        }
+        // update as a normal module
 
-            return true;
+        // TODO: A latest packet is always sent even after the base is fully built, we don't want that
+        // When #2021 is merged, just use ThrottledPacketSender.RemovePendingPackets
+        Resolve<ThrottledPacketSender>().SendThrottled(new ModifyConstructedAmount(entityId, amount),
+            (packet) => { return packet.GhostId; }, 0.1f);
+    }
+
+    public static IEnumerator BroadcastObjectBuilt(ConstructableBase constructableBase, NitroxId entityId)
+    {
+        BaseGhost baseGhost = constructableBase.model.GetComponent<BaseGhost>();
+        constructableBase.SetState(true, true);
+        if (!baseGhost.targetBase)
+        {
+            Log.Error("Something wrong happened, couldn't find base after finishing building ghost");
+            yield break;
+        }
+        Base targetBase = baseGhost.targetBase;
+        Base parentBase = null;
+        if (constructableBase.tr.parent)
+        {
+            parentBase = constructableBase.GetComponentInParent<Base>(true);
         }
 
-        public static void Postfix(Constructable __instance, bool __result)
+        // If a module was spawned we need to transfer the ghost id to it for further recognition
+        BuildUtils.TryTransferIdFromGhostToModule(baseGhost, entityId, constructableBase, out GameObject moduleObject);
+
+        // Have a delay for baseGhost to be actually destroyed
+        yield return null;
+
+        if (parentBase)
         {
-            if (!__result) return;
+            // update existing base
+            if (!parentBase.TryGetNitroxId(out NitroxId parentId))
+            {
+                // TODO: Probably add a resync here
```

Comment:
Shouldn't that be catched by the failedOperationä-tracker which encourages the user to resync?

Reply:
The issue here is you don't know what's the id of the broken piece so you can at best at a "general desync count" but that's all.